apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -107,56 +107,71 @@ defmodule OMG.Watcher.BlockGetter do
     {:noreply, state}
   end
 
-  @decorate measure_start()
-  def handle_cast(
-        {:apply_block,
-         %BlockApplication{
-           transactions: transactions,
-           number: blknum,
-           eth_height: eth_height
-         } = to_apply},
-        state
-      ) do
-    with {:ok, _} <- Core.chain_ok(state),
-         tx_exec_results = for(tx <- transactions, do: OMG.State.exec(tx, :ignore)),
-         {:ok, state} <- Core.validate_executions(tx_exec_results, to_apply, state) do
-      _ =
-        to_apply
+  def handle_continue({:execute_transactions, block_application}, state) do
+    tx_exec_results = for(tx <- block_application.transactions, do: OMG.State.exec(tx, :ignore))
+
+    case Core.validate_executions(tx_exec_results, block_application, state) do
+      {:ok, state} ->
+        block_application
         |> Core.ensure_block_imported_once(state)
         |> Enum.each(&DB.Transaction.update_with/1)
 
-      state = run_block_download_task(state)
+        {:noreply, state, {:continue, {:run_block_download_task, block_application}}}
```

Comment:
Have you forgotten to delete it?

Reply:
what do you mean?

Reply:
my mistake, 
I did not see the indentation